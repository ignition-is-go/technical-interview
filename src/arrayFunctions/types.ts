export interface Person {
	name: string,
	netWorth: number,
	coder?: boolean,
	us?: boolean
	city: string,
	languages: string[]
}

export const cities = ['nyc', 'sf', 'la']
export const languages = [
	'javascript',
	'typescript',
	'html',
	'css',
	'c#',
	'python',
	'ruby',
]
