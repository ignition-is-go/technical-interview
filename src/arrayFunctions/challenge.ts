import { Person } from './types'

/**
 * Return an array of all the people in the provided array who know typescript
 * @param people an array of Person type objects. See ./types for person description
 */

export function allCodersWhoKnowTypescript(people: Person[]): Person[] {

	return people.filter(p => p.languages.includes('typescript'))
}

export function objectWithPeopleByCity(people: Person[]): object {

	return people.reduce((cities, person) => {
		return {
			...cities,
			[person.city]: [
				...(Reflect.has(cities, person.city) ? cities[person.city] : []),
				person,
			],
		}
	}, {})
}

export function objectWithCountsOfNumberOfDevelopersWhoKnowEachLanguage(people: Person[]): object {

	return people.reduce((languages, person) => {
		person.languages.forEach(l => languages[l] ? languages[l] += 1 : languages[l] = 1)
		return languages
	}, {})
}

export function codersSortedByNetWorth(people: Person[]): Person[] {
	return people.filter(p => p.coder).sort((a, b) => a.netWorth - b.netWorth)
}
