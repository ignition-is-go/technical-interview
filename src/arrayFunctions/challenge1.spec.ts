// tslint:disable: no-console
import * as R from 'ramda'
import { allCodersWhoKnowTypescript, codersSortedByNetWorth, objectWithCountsOfNumberOfDevelopersWhoKnowEachLanguage, objectWithPeopleByCity } from './challenge'
import { Person } from './types'

const people = require('./data/people.json') as Person[]

describe('all coders who know typescript', () => {
	let func, safePeople
	beforeEach(() => {
		safePeople = R.clone(people)
		func = () => allCodersWhoKnowTypescript(safePeople)
	})

	it('returns an array with the correct answer', () => {
		const result = func()

		expect(Array.isArray(result)).toBeTruthy()
		expect(result.length).toBeGreaterThan(0)

		expect(result).toMatchSnapshot()
		result.forEach(p => {
			expect(p.coder).toBeTruthy()
			expect(p.languages).toContain('typescript')
		})
	})

	it('runs quickly', () => {
		expect(time(func)).toBeLessThan(.2)
	})

	it('does not throw errors', () => {
		expect(func).not.toThrow()
	})

	it('does not modify its arguments', () => {
		func()
		expect(safePeople).toMatchObject(people)
	})
})

describe('object with people by city', () => {
	let func, safePeople
	beforeEach(() => {
		safePeople = R.clone(people)
		func = () => objectWithPeopleByCity(safePeople)
	})

	it('returns an array with the correct answer', () => {
		const result = func()
		expect(typeof result).toBe('object')
		expect(result).toMatchSnapshot()
		Reflect.ownKeys(result).forEach((city) => {
			const cityPeople = result[city]
			cityPeople.forEach(p => expect(p.city).toMatch(city as string))
		})
	})

	it('runs quickly', () => {
		expect(time(func)).toBeLessThan(1)
	})

	it('does not throw errors', () => {
		expect(func).not.toThrow()
	})

	it('does not modify its arguments', () => {
		func()
		expect(safePeople).toMatchObject(people)
	})
})

describe('object with counts of nubmer of developers who know each language', () => {
	let func, safePeople
	beforeEach(() => {
		safePeople = R.clone(people)
		func = () => objectWithCountsOfNumberOfDevelopersWhoKnowEachLanguage(safePeople)
	})

	it('returns an array with the correct answer', () => {
		const result = func()
		expect(typeof result).toBe('object')
		expect(result).toMatchSnapshot()
		Reflect.ownKeys(result).forEach((language) => {
			const count = result[language]
			expect(typeof count).toBe('number')
		})
	})

	it('runs quickly', () => {
		expect(time(func)).toBeLessThan(.2)
	})

	it('does not throw errors', () => {
		expect(func).not.toThrow()
	})

	it('does not modify its arguments', () => {
		func()
		expect(safePeople).toMatchObject(people)
	})
})

describe('codersSortedByNetWorth', () => {
	let func, safePeople
	beforeEach(() => {
		safePeople = R.clone(people)
		func = () => codersSortedByNetWorth(safePeople)
	})

	it('returns an array with the correct answer', () => {
		const result = func()
		expect(Array.isArray(result)).toBeTruthy()
		expect(result).toMatchSnapshot()
		result.forEach((coder: Person, idx) => {
			expect(coder.coder).toBeTruthy()
			if (idx > 0) {
				expect(coder.netWorth).toBeGreaterThanOrEqual(result[idx - 1].netWorth)
			}
			if (idx < result.length - 1) {
				expect(coder.netWorth).toBeLessThan(result[idx + 1].netWorth)
			}
		})
	})

	it('runs quickly', () => {
		expect(time(func)).toBeLessThan(.2)
	})

	it('does not throw errors', () => {
		expect(func).not.toThrow()
	})

	it('does not modify its arguments', () => {
		func()
		expect(safePeople).toMatchObject(people)
	})
})

const time = (func: () => any) => {
	const start = process.hrtime()
	func()
	return process.hrtime(start)[1] / 1000000
}
