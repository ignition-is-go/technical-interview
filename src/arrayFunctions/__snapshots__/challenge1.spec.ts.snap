// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all coders who know typescript returns an array with the correct answer 1`] = `
Array [
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "html",
      "typescript",
      "c#",
    ],
    "name": "Claudie Gleichner",
    "netWorth": 37056,
  },
  Object {
    "city": "la",
    "coder": true,
    "languages": Array [
      "ruby",
      "c#",
      "typescript",
    ],
    "name": "Mr. Lee Cartwright",
    "netWorth": 89490,
    "us": false,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "typescript",
      "css",
      "html",
      "python",
    ],
    "name": "Brandy Dicki",
    "netWorth": 98959,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "python",
      "c#",
      "typescript",
    ],
    "name": "Junius Anderson",
    "netWorth": 85224,
    "us": true,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "typescript",
      "css",
    ],
    "name": "Esperanza Kilback",
    "netWorth": 28749,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "html",
      "c#",
      "typescript",
      "css",
      "python",
    ],
    "name": "Oswald Green",
    "netWorth": 64182,
    "us": true,
  },
]
`;

exports[`codersSortedByNetWorth returns an array with the correct answer 1`] = `
Array [
  Object {
    "city": "la",
    "coder": true,
    "languages": Array [],
    "name": "Carol Schroeder",
    "netWorth": 7066,
    "us": true,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "ruby",
      "css",
      "html",
      "python",
      "c#",
    ],
    "name": "Priscilla Funk",
    "netWorth": 13840,
    "us": true,
  },
  Object {
    "city": "la",
    "coder": true,
    "languages": Array [
      "css",
      "python",
    ],
    "name": "Hugh Hayes Jr.",
    "netWorth": 17191,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "c#",
    ],
    "name": "Judah Welch III",
    "netWorth": 19164,
    "us": true,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "typescript",
      "css",
    ],
    "name": "Esperanza Kilback",
    "netWorth": 28749,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "ruby",
    ],
    "name": "Tony Bode",
    "netWorth": 31694,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "html",
      "typescript",
      "c#",
    ],
    "name": "Claudie Gleichner",
    "netWorth": 37056,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "css",
      "html",
    ],
    "name": "Maribel Jaskolski",
    "netWorth": 44793,
    "us": false,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "c#",
    ],
    "name": "Creola Hayes",
    "netWorth": 48876,
    "us": false,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "css",
      "python",
    ],
    "name": "Jannie Hessel III",
    "netWorth": 56512,
    "us": true,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "html",
      "c#",
      "typescript",
      "css",
      "python",
    ],
    "name": "Oswald Green",
    "netWorth": 64182,
    "us": true,
  },
  Object {
    "city": "la",
    "coder": true,
    "languages": Array [
      "css",
      "ruby",
      "c#",
    ],
    "name": "Junius Armstrong",
    "netWorth": 71316,
  },
  Object {
    "city": "la",
    "coder": true,
    "languages": Array [
      "html",
      "ruby",
      "c#",
    ],
    "name": "Alexane Kuvalis IV",
    "netWorth": 73170,
  },
  Object {
    "city": "la",
    "coder": true,
    "languages": Array [
      "css",
      "python",
    ],
    "name": "Trisha Leuschke",
    "netWorth": 83162,
  },
  Object {
    "city": "la",
    "coder": true,
    "languages": Array [
      "c#",
    ],
    "name": "Conor Casper",
    "netWorth": 84946,
    "us": true,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "python",
      "c#",
      "typescript",
    ],
    "name": "Junius Anderson",
    "netWorth": 85224,
    "us": true,
  },
  Object {
    "city": "la",
    "coder": true,
    "languages": Array [
      "ruby",
      "c#",
    ],
    "name": "Ruth Kunze",
    "netWorth": 88295,
    "us": false,
  },
  Object {
    "city": "la",
    "coder": true,
    "languages": Array [
      "ruby",
      "c#",
      "typescript",
    ],
    "name": "Mr. Lee Cartwright",
    "netWorth": 89490,
    "us": false,
  },
  Object {
    "city": "la",
    "coder": true,
    "languages": Array [
      "css",
      "c#",
      "ruby",
    ],
    "name": "Ollie O'Hara",
    "netWorth": 93045,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "css",
      "html",
      "ruby",
    ],
    "name": "Miss Manley Schaefer",
    "netWorth": 96702,
  },
  Object {
    "city": "sf",
    "coder": true,
    "languages": Array [
      "typescript",
      "css",
      "html",
      "python",
    ],
    "name": "Brandy Dicki",
    "netWorth": 98959,
  },
]
`;

exports[`object with counts of nubmer of developers who know each language returns an array with the correct answer 1`] = `
Object {
  "c#": 12,
  "css": 11,
  "html": 7,
  "python": 7,
  "ruby": 8,
  "typescript": 6,
}
`;

exports[`object with people by city returns an array with the correct answer 1`] = `
Object {
  "la": Array [
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Torrance Kuhic",
      "netWorth": 90995,
      "us": true,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Ms. Jacques Herman",
      "netWorth": 3859,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Mollie Runolfsson",
      "netWorth": 54329,
      "us": true,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Makayla Anderson",
      "netWorth": 84151,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Letitia Ortiz",
      "netWorth": 95775,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Elwin Nienow",
      "netWorth": 88507,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Laurel Rogahn",
      "netWorth": 56561,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Levi Padberg",
      "netWorth": 30515,
      "us": true,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Natalie Hammes",
      "netWorth": 56055,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Ewell Powlowski",
      "netWorth": 27396,
      "us": true,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Herminio Tremblay",
      "netWorth": 11177,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Terrance Bartell",
      "netWorth": 60249,
    },
    Object {
      "city": "la",
      "coder": true,
      "languages": Array [
        "html",
        "ruby",
        "c#",
      ],
      "name": "Alexane Kuvalis IV",
      "netWorth": 73170,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Dr. Name Kuphal",
      "netWorth": 15099,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Fausto Brakus",
      "netWorth": 24427,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Shawn Langosh",
      "netWorth": 84461,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Florencio Bednar",
      "netWorth": 97691,
      "us": true,
    },
    Object {
      "city": "la",
      "coder": true,
      "languages": Array [
        "css",
        "c#",
        "ruby",
      ],
      "name": "Ollie O'Hara",
      "netWorth": 93045,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Dallas Rosenbaum",
      "netWorth": 64880,
      "us": true,
    },
    Object {
      "city": "la",
      "coder": true,
      "languages": Array [
        "c#",
      ],
      "name": "Conor Casper",
      "netWorth": 84946,
      "us": true,
    },
    Object {
      "city": "la",
      "coder": true,
      "languages": Array [
        "ruby",
        "c#",
        "typescript",
      ],
      "name": "Mr. Lee Cartwright",
      "netWorth": 89490,
      "us": false,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Aleen Cremin",
      "netWorth": 10082,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Kenyatta Boyer MD",
      "netWorth": 81343,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Mollie Mitchell",
      "netWorth": 21461,
      "us": false,
    },
    Object {
      "city": "la",
      "coder": true,
      "languages": Array [
        "css",
        "python",
      ],
      "name": "Hugh Hayes Jr.",
      "netWorth": 17191,
    },
    Object {
      "city": "la",
      "coder": true,
      "languages": Array [],
      "name": "Carol Schroeder",
      "netWorth": 7066,
      "us": true,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Celia Ruecker",
      "netWorth": 21387,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Betty Spencer",
      "netWorth": 79495,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Linwood Rowe",
      "netWorth": 67842,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Alexandria Kunze",
      "netWorth": 50671,
      "us": true,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Kaci Hansen",
      "netWorth": 90622,
      "us": true,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Devonte Lynch",
      "netWorth": 81087,
      "us": true,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Milton Wolff",
      "netWorth": 24054,
      "us": false,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Aiden Collier V",
      "netWorth": 9132,
      "us": false,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Cassie Veum",
      "netWorth": 74397,
      "us": false,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Miss Hiram Larkin",
      "netWorth": 85997,
      "us": true,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Kameron Johns",
      "netWorth": 81838,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Gregorio Stamm",
      "netWorth": 76238,
      "us": true,
    },
    Object {
      "city": "la",
      "coder": true,
      "languages": Array [
        "ruby",
        "c#",
      ],
      "name": "Ruth Kunze",
      "netWorth": 88295,
      "us": false,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Brenden Hoeger",
      "netWorth": 60219,
      "us": true,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Euna Marvin",
      "netWorth": 75615,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Coralie Olson",
      "netWorth": 44230,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Jessyca Runte III",
      "netWorth": 41508,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Desmond Ernser",
      "netWorth": 79608,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Eric Vandervort",
      "netWorth": 80370,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Valerie Rutherford",
      "netWorth": 70192,
      "us": true,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Allison Lubowitz",
      "netWorth": 18394,
      "us": false,
    },
    Object {
      "city": "la",
      "coder": false,
      "languages": Array [],
      "name": "Mrs. Lempi Bruen",
      "netWorth": 94560,
    },
    Object {
      "city": "la",
      "coder": true,
      "languages": Array [
        "css",
        "python",
      ],
      "name": "Trisha Leuschke",
      "netWorth": 83162,
    },
    Object {
      "city": "la",
      "languages": Array [],
      "name": "Mrs. Percival Kuhic",
      "netWorth": 6492,
      "us": false,
    },
    Object {
      "city": "la",
      "coder": true,
      "languages": Array [
        "css",
        "ruby",
        "c#",
      ],
      "name": "Junius Armstrong",
      "netWorth": 71316,
    },
  ],
  "sf": Array [
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Dr. Aniyah Walker",
      "netWorth": 14403,
    },
    Object {
      "city": "sf",
      "coder": false,
      "languages": Array [],
      "name": "Kenny Hamill",
      "netWorth": 13722,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Mrs. Murphy Kozey",
      "netWorth": 33914,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Merlin Carter",
      "netWorth": 33813,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Quinn Kling",
      "netWorth": 99931,
      "us": true,
    },
    Object {
      "city": "sf",
      "coder": false,
      "languages": Array [],
      "name": "Timmothy Runolfsdottir",
      "netWorth": 97681,
    },
    Object {
      "city": "sf",
      "coder": true,
      "languages": Array [
        "html",
        "typescript",
        "c#",
      ],
      "name": "Claudie Gleichner",
      "netWorth": 37056,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Marcel Rolfson",
      "netWorth": 5358,
      "us": true,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Henriette Turner",
      "netWorth": 94586,
      "us": true,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Mrs. Nathanael Predovic",
      "netWorth": 5705,
      "us": true,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Gerda Glover",
      "netWorth": 80266,
    },
    Object {
      "city": "sf",
      "coder": false,
      "languages": Array [],
      "name": "Monica Schowalter",
      "netWorth": 31221,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Naomi Borer",
      "netWorth": 42939,
    },
    Object {
      "city": "sf",
      "coder": false,
      "languages": Array [],
      "name": "Addison Breitenberg",
      "netWorth": 35351,
    },
    Object {
      "city": "sf",
      "coder": true,
      "languages": Array [
        "ruby",
        "css",
        "html",
        "python",
        "c#",
      ],
      "name": "Priscilla Funk",
      "netWorth": 13840,
      "us": true,
    },
    Object {
      "city": "sf",
      "coder": false,
      "languages": Array [],
      "name": "Oliver Zboncak Sr.",
      "netWorth": 94222,
      "us": true,
    },
    Object {
      "city": "sf",
      "coder": true,
      "languages": Array [
        "css",
        "html",
      ],
      "name": "Maribel Jaskolski",
      "netWorth": 44793,
      "us": false,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Russel Corkery",
      "netWorth": 77955,
    },
    Object {
      "city": "sf",
      "coder": true,
      "languages": Array [
        "css",
        "python",
      ],
      "name": "Jannie Hessel III",
      "netWorth": 56512,
      "us": true,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Tyshawn Ferry",
      "netWorth": 42052,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Dexter Streich",
      "netWorth": 20899,
      "us": true,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Darlene Bins",
      "netWorth": 62826,
    },
    Object {
      "city": "sf",
      "coder": false,
      "languages": Array [],
      "name": "Hilbert Boyer",
      "netWorth": 30746,
      "us": false,
    },
    Object {
      "city": "sf",
      "coder": false,
      "languages": Array [],
      "name": "Miss Jalon Huels",
      "netWorth": 96392,
      "us": false,
    },
    Object {
      "city": "sf",
      "coder": true,
      "languages": Array [
        "typescript",
        "css",
        "html",
        "python",
      ],
      "name": "Brandy Dicki",
      "netWorth": 98959,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Abbigail Von",
      "netWorth": 59809,
      "us": false,
    },
    Object {
      "city": "sf",
      "coder": false,
      "languages": Array [],
      "name": "Karson Bosco",
      "netWorth": 82758,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Norwood Bechtelar",
      "netWorth": 67764,
      "us": false,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Dr. Jacinthe Green",
      "netWorth": 94213,
    },
    Object {
      "city": "sf",
      "coder": false,
      "languages": Array [],
      "name": "Mr. Kadin Wiegand",
      "netWorth": 65315,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Ms. Alysa Weber",
      "netWorth": 95703,
      "us": false,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Aniyah Kuvalis",
      "netWorth": 48523,
    },
    Object {
      "city": "sf",
      "coder": true,
      "languages": Array [
        "c#",
      ],
      "name": "Judah Welch III",
      "netWorth": 19164,
      "us": true,
    },
    Object {
      "city": "sf",
      "coder": true,
      "languages": Array [
        "css",
        "html",
        "ruby",
      ],
      "name": "Miss Manley Schaefer",
      "netWorth": 96702,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Coty Wintheiser",
      "netWorth": 93871,
      "us": false,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Kyle Hauck",
      "netWorth": 58129,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Carter Mayert",
      "netWorth": 49960,
      "us": true,
    },
    Object {
      "city": "sf",
      "coder": true,
      "languages": Array [
        "python",
        "c#",
        "typescript",
      ],
      "name": "Junius Anderson",
      "netWorth": 85224,
      "us": true,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Chauncey Herzog DDS",
      "netWorth": 86856,
      "us": true,
    },
    Object {
      "city": "sf",
      "coder": true,
      "languages": Array [
        "typescript",
        "css",
      ],
      "name": "Esperanza Kilback",
      "netWorth": 28749,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "David Ratke",
      "netWorth": 98604,
      "us": false,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Eliezer Lindgren I",
      "netWorth": 61728,
    },
    Object {
      "city": "sf",
      "coder": true,
      "languages": Array [
        "html",
        "c#",
        "typescript",
        "css",
        "python",
      ],
      "name": "Oswald Green",
      "netWorth": 64182,
      "us": true,
    },
    Object {
      "city": "sf",
      "coder": true,
      "languages": Array [
        "c#",
      ],
      "name": "Creola Hayes",
      "netWorth": 48876,
      "us": false,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Noe Buckridge",
      "netWorth": 35390,
    },
    Object {
      "city": "sf",
      "coder": true,
      "languages": Array [
        "ruby",
      ],
      "name": "Tony Bode",
      "netWorth": 31694,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Bryana Boehm",
      "netWorth": 33718,
      "us": true,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Maxine Franecki",
      "netWorth": 6643,
      "us": false,
    },
    Object {
      "city": "sf",
      "languages": Array [],
      "name": "Juvenal Crist",
      "netWorth": 34562,
      "us": true,
    },
  ],
}
`;
